File: core/src/test/java/net/dontdrinkandroot/wicket/behavior/DisplayNoneAppenderTest.java
package net.dontdrinkandroot.wicket.behavior;

import net.dontdrinkandroot.wicket.AbstractWicketTest;
import org.apache.wicket.core.util.string.ComponentRenderer;
import org.apache.wicket.markup.html.WebMarkupContainer;
import org.junit.Assert;
import org.junit.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class DisplayNoneAppenderTest extends AbstractWicketTest
{
    @Test
    public void testModelConstructor()
    {
        WebMarkupContainer container = new WebMarkupContainer("id");
        container.add(new DisplayNoneAppender());
        CharSequence componentMarkup = ComponentRenderer.renderComponent(container);

        Assert.assertEquals(
                "<wicket:container wicket:id=\"id\" style=\"display: none\"></wicket:container>",
                componentMarkup.toString()
        );
    }
}

File: core/src/test/kotlin/net/dontdrinkandroot/wicket/behavior/DisplayNoneAppenderTest.kt
package net.dontdrinkandroot.wicket.behavior

import net.dontdrinkandroot.wicket.test.AbstractWicketTest
import org.apache.wicket.core.util.string.ComponentRenderer
import org.apache.wicket.markup.html.WebMarkupContainer
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class DisplayNoneAppenderTest : AbstractWicketTest() {

    @Test
    fun testModelConstructor() {
        val container = WebMarkupContainer("id")
        container.add(DisplayNoneAppender())
        val componentMarkup = ComponentRenderer.renderComponent(container)
        Assertions.assertEquals(
            "<wicket:container wicket:id=\"id\" style=\"display: none\"></wicket:container>",
            componentMarkup.toString()
        )
    }
}